fence/sync/sync_users.py

```diff
@@ -966,24 +970,48 @@ def sync(self):
             with self.driver.session as s:
                 self._sync(s)
 
+    def download(self):
+        if self.session:
+            self._download(self.session)
+        else:
+            with self.driver.session as s:
+                self._download(s)
+
+    def _download(self, sess):
+        """
+        Download files from dbgap server.
+        """
+        folderdir = os.path.join("/mnt/shared/", str(self.folder))
+
+        if not os.path.exists(folderdir):
+            os.makedirs(folderdir)
+
+        self.logger.info("Download from server")
+        try:
+            if self.protocol == "sftp":
+                self._get_from_sftp_with_proxy(folderdir)
+            else:
+                self._get_from_ftp_with_proxy(folderdir)
+            dbgap_file_list = glob.glob(os.path.join(folderdir, "*"))
+            return dbgap_file_list
+        except Exception as e:
+            self.logger.error(e)
+            exit(1)
+
     def _sync(self, sess):
         """
         Collect files from dbgap server, sync csv and yaml files to storage
         backend and fence DB
         """
         dbgap_file_list = []
-        tmpdir = tempfile.mkdtemp()
+        folderdir = os.path.join("/mnt/shared/", str(self.folder))
+
         if self.is_sync_from_dbgap_server:
-            self.logger.info("Download from server")
-            try:
-                if self.protocol == "sftp":
-                    self._get_from_sftp_with_proxy(tmpdir)
-                else:
-                    self._get_from_ftp_with_proxy(tmpdir)
-                dbgap_file_list = glob.glob(os.path.join(tmpdir, "*"))
-            except Exception as e:
-                self.logger.error(e)
-                exit(1)
+            if self.folder and os.path.exists(folderdir):
```

Comment:
`self.folder` has a default value so it will never be None
```suggestion
            if os.path.exists(folderdir):
```